chat.quadrix.Quadrix.yml

```diff
@@ -0,0 +1,62 @@
+app-id: chat.quadrix.Quadrix
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-download
+build-options:
+  append-path: /usr/lib/sdk/node16/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: quadrix
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/quadrix/flatpak-node/cache
+        npm_config_cache: /run/build/quadrix/flatpak-node/npm-cache
+        npm_config_nodedir: /usr/lib/sdk/node16
+        npm_config_offline: 'true'
+    subdir: main
+    sources:
+      - type: dir
+        path: ..
+        dest: main
```

Comment:
so where does it get the sources?

I missed that on first review, but that's the reason the build failed.

Reply:
Hmmm, the source is on my local machine. I was wondering what was being built by the flathubbot. Obviously the manifest as it is cannot be enough to build the app.

Does it mean I cannot upload a local build to flathub directly, and flathub has to build the whole app by itself? Is it something specific to Electron apps? Would it be easier to use a .deb file as the source?

Reply:
The manifest is to build the flatpak and this is done on flathub infrastructure.

You can put a source as git, a tarball, etc. as per the documentation.